docker/Dockerfile.base

```diff
@@ -22,28 +22,54 @@ RUN mkdir -p /kb/deployment/services/narrative /tmp/narrative
 
 # Add the R CRAN repo - add this back in when the 3.1-beta issue is resolved
 RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E084DAB9
-RUN echo 'deb http://cran.cnr.berkeley.org/bin/linux/ubuntu trusty/' | tee /etc/apt/sources.list.d/cran.list
-
-RUN DEBIAN_FRONTEND=noninteractive apt-get update
-
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y mesa-common-dev libglu1-mesa-dev
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y r-recommended
+RUN echo 'deb http://cran.cnr.berkeley.edu/bin/linux/ubuntu trusty/' | tee /etc/apt/sources.list.d/cran.list
 
 # There are a bunch of packages that are related to kernel operation
 # which we can't upgrade within a container. Mark them for "hold" before
 # running the upgrade
-RUN DEBIAN_FRONTEND=noninteractive apt-mark hold udev initscripts plymouth initramfs-tools procps busybox-initramfs
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
+apt-mark hold udev initscripts plymouth initramfs-tools procps busybox-initramfs
 RUN DEBIAN_FRONTEND=noninteractive apt-get -f upgrade -y
 
 # Install the stuff we actually need
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y perl-base perl-modules make
-
-# Maybe this will get all the necessary dependencies!
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y python-qt4 python-pip python-setuptools ipython-notebook python-matplotlib python-dev python-scipy python-numpy python-lxml python-sklearn python-sympy python-pandas python-rpy2 python-virtualenv
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y libfreetype6-dev libcairo2-dev texlive-latex-base texlive-fonts-recommended curl libcurl4-gnutls-dev git-core libzmq-dev libreadline6-dev libreadline-dev libssl-dev libxml2-dev libxslt1-dev
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y \
+perl-base \
+perl-modules \
+make \
+python-qt4 \
+python-pip \
+python-setuptools \
+ipython-notebook \
+python-matplotlib \
+python-dev \
+python-scipy \
+python-numpy \
+python-lxml \
+python-sklearn \
+python-sympy \
+python-pandas \
+python-rpy2 \
+python-virtualenv \
+python-thrift \
+python-yaml \
+python-paramiko \
+libfreetype6-dev \
+libcairo2-dev \
+texlive-latex-base \
+texlive-fonts-recommended \
+curl \
+libcurl4-gnutls-dev \
+git-core \
+libzmq-dev \
+libreadline6-dev \
+libreadline-dev \
+libssl-dev \
+libxml2-dev \
+libxslt1-dev
 
-# There's a bug in the current repository deployment of python requests - do an easy_install of v2.3.0, or pip will fail
-RUN easy_install requests==2.3.0
+# Pip is currently broken when installed through Ubuntu. Upgrade it with easy_install 
+RUN easy_install -U pip
```

Comment:
I believe so. It might be an issue with the pip that lives in Ubuntu trusty mirrors. Either way, the best advice I found was to upgrade pip with easy_install.